numba/experimental/jitclass/boxing.py

```diff
@@ -144,12 +144,14 @@ def _specialize_box(typ):
         "__ixor__",
     }
     for name, func in typ.methods.items():
+        if name == "__init__":
+            continue
         if (
             name.startswith("__")
             and name.endswith("__")
             and name not in supported_dunders
         ):
-            continue
+            raise TypeError("Method {0} is not supported".format(name))
```

Comment:
```suggestion
            raise TypeError(f"Method '{name}' is not supported.")
```
suggest fstring and also putting the name into quotes such that it reads e.g.:
```
Method '__enter__' is not supported.
```